libsplinter/src/admin/service/mod.rs

```diff
@@ -325,6 +331,60 @@ impl Service for AdminService {
     }
 }
 
+#[derive(Clone)]
+struct AdminServiceCommands {
+    shared: Arc<Mutex<AdminServiceShared>>,
+}
+
+impl AdminCommands for AdminServiceCommands {
+    fn submit_circuit_change(
+        &self,
+        circuit_change: CircuitManagementPayload,
+    ) -> Result<(), AdminServiceError> {
+        self.shared
+            .lock()
+            .map_err(|_| AdminServiceError::general_error("Admin shared lock was Lock poisoned"))?
```

Comment:
Is the capital Lock intentional?  "Admin shared lock was Lock poisoned"

Reply:
Corrected.